plugins/utils/todo-comments.nix

```diff
@@ -16,275 +24,400 @@ let
 
-        multilineContext = helpers.defaultNullOpts.mkInt 10 ''
-          Extra lines that will be re-evaluated when changing a line.
-        '';
+    gui_style = {
+      fg = defaultNullOpts.mkStr "NONE" ''
+        The gui style to use for the fg highlight group.
+      '';
 
-        before = helpers.defaultNullOpts.mkStr "" ''
-          "fg" or "bg" or empty.
-        '';
+      bg = defaultNullOpts.mkStr "BOLD" ''
+        The gui style to use for the bg highlight group.
+      '';
+    };
 
-        keyword = helpers.defaultNullOpts.mkStr "wide" ''
-          "fg", "bg", "wide", "wide_bg", "wide_fg" or empty.
-          (wide and wide_bg is the same as bg, but will also highlight
-          surrounding characters, wide_fg acts accordingly but with fg).
-        '';
+    merge_keywords = defaultNullOpts.mkBool true ''
+      When true, custom keywords will be merged with the default.
+    '';
 
-        after = helpers.defaultNullOpts.mkStr "fg" ''
-          "fg" or "bg" or empty.
-        '';
+    highlight = {
+      multiline = defaultNullOpts.mkBool true ''
+        Enable multiline todo comments.
+      '';
 
-        pattern =
-          helpers.defaultNullOpts.mkNullable (with types; either str (listOf str)) ".*<(KEYWORDS)\\s*:"
-            ''
-              Pattern or list of patterns, used for highlighting (vim regex)
+      multiline_pattern = defaultNullOpts.mkStr "^." ''
+        Lua pattern to match the next multiline from the start of the
+        matched keyword.
+      '';
 
-              Note: the provided pattern will be embedded as such: `[[PATTERN]]`.
-            '';
+      multiline_context = defaultNullOpts.mkInt 10 ''
+        Extra lines that will be re-evaluated when changing a line.
+      '';
 
-        commentsOnly = helpers.defaultNullOpts.mkBool true ''
-          Uses treesitter to match keywords in comments only.
-        '';
+      before = defaultNullOpts.mkStr "" ''
+        "fg" or "bg" or empty.
+      '';
 
-        maxLineLen = helpers.defaultNullOpts.mkInt 400 ''
-          Ignore lines longer than this.
-        '';
+      keyword = defaultNullOpts.mkStr "wide" ''
+        "fg", "bg", "wide", "wide_bg", "wide_fg" or empty.
+        (wide and wide_bg is the same as bg, but will also highlight
+        surrounding characters, wide_fg acts accordingly but with fg).
+      '';
 
-        exclude = helpers.mkNullOrOption (types.listOf types.str) ''
-          List of file types to exclude highlighting.
+      after = defaultNullOpts.mkStr "fg" ''
+        "fg" or "bg" or empty.
+      '';
+
+      pattern = defaultNullOpts.mkNullable' {
+        # TODO: Added 2024-08-16, remove after 24.11
+        # Before we modified value automatically turning it into a raw regex.
+        type = transitionType (with types; either str (listOf str)) (
+          text: if isList text then map (elem: mkRaw "[[${elem}]]") text else mkRaw "[[${text}]]"
+        ) (with types; either rawLua (listOf rawLua));
```

Comment:
I think transition type would be confused by the composite type (listOf). Especially for empty list definitions.

Perhaps we could do `either (transitionType str) (listOf (transitionType str))`?